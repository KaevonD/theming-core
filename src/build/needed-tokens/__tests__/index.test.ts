// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import findNeededTokens from '..';
import { join } from 'path';
import { variablesMap } from './__fixtures__/variables-map';

const scssDir = join(__dirname, '__fixtures__', 'scss');

describe('findNeededTokens returns correct tokens', () => {
  test('without exposed tokens, without variablesMap', () => {
    const ouput = findNeededTokens(scssDir, {}, []);
    expect(ouput).toEqual([]);
  });

  test('without exposed tokens, with variablesMap', () => {
    const ouput = findNeededTokens(scssDir, variablesMap, []);
    expect(ouput).toMatchSnapshot();
  });

  test('with exposed tokens, with variablesMap', () => {
    const ouput = findNeededTokens(scssDir, variablesMap, ['colorbackgroundbuttonnormaldisabled', 'additionalExposed']);
    expect(ouput).toMatchSnapshot();
  });
});
