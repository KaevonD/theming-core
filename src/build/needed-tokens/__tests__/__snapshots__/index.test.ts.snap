// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findNeededTokens returns correct tokens with exposed tokens, with variablesMap 1`] = `
Array [
  "borderalertradius",
  "borderfieldwidth",
  "colorbackgroundbuttonnormalactive",
  "colorbackgroundbuttonnormaldefault",
  "colorbackgroundbuttonnormaldisabled",
  "colorbackgroundbuttonnormalhover",
  "colorbackgroundcontainercontent",
  "colorbackgroundstatuserror",
  "colorbackgroundstatusinfo",
  "colorbackgroundstatussuccess",
  "colorbackgroundstatuswarning",
  "colorborderbuttonnormalactive",
  "colorborderbuttonnormaldefault",
  "colorborderbuttonnormaldisabled",
  "colorborderbuttonnormalhover",
  "colorborderstatuserror",
  "colorborderstatusinfo",
  "colorborderstatussuccess",
  "colorborderstatuswarning",
  "colortextbuttonnormalactive",
  "colortextbuttonnormaldefault",
  "colortextbuttonnormalhover",
  "colortextinteractivedisabled",
  "colortextstatuserror",
  "colortextstatusinfo",
  "colortextstatussuccess",
  "colortextstatuswarning",
  "spacealertactionleft",
  "spacealerthorizontal",
  "spacealertmessageright",
  "spacealertvertical",
  "spaces",
  "spacescaledxxs",
  "spacexxs",
  "additionalExposed",
]
`;

exports[`findNeededTokens returns correct tokens without exposed tokens, with variablesMap 1`] = `
Array [
  "borderalertradius",
  "borderfieldwidth",
  "colorbackgroundbuttonnormalactive",
  "colorbackgroundbuttonnormaldefault",
  "colorbackgroundbuttonnormaldisabled",
  "colorbackgroundbuttonnormalhover",
  "colorbackgroundcontainercontent",
  "colorbackgroundstatuserror",
  "colorbackgroundstatusinfo",
  "colorbackgroundstatussuccess",
  "colorbackgroundstatuswarning",
  "colorborderbuttonnormalactive",
  "colorborderbuttonnormaldefault",
  "colorborderbuttonnormaldisabled",
  "colorborderbuttonnormalhover",
  "colorborderstatuserror",
  "colorborderstatusinfo",
  "colorborderstatussuccess",
  "colorborderstatuswarning",
  "colortextbuttonnormalactive",
  "colortextbuttonnormaldefault",
  "colortextbuttonnormalhover",
  "colortextinteractivedisabled",
  "colortextstatuserror",
  "colortextstatusinfo",
  "colortextstatussuccess",
  "colortextstatuswarning",
  "spacealertactionleft",
  "spacealerthorizontal",
  "spacealertmessageright",
  "spacealertvertical",
  "spaces",
  "spacescaledxxs",
  "spacexxs",
]
`;
